.pipelines/singletenancy/aks/e2e-job-template.yaml

```diff
@@ -1,44 +1,98 @@
 parameters:
   name: ""
   displayName: ""
-  arch: ""  
+  arch: ""
   os: ""
   clusterType: ""
   clusterName: ""
+  vmSize: ""
   k8sVersion: ""
   windowsOsSku: ""
   os_version: ""
+  scaleup: ""
 
 stages:
   - stage: ${{ parameters.name }}
     displayName: E2E - ${{ parameters.displayName }}
-    dependsOn: 
+    variables:
+      GOPATH: "$(Agent.TempDirectory)/go" # Go workspace path
+      GOBIN: "$(GOPATH)/bin" # Go binaries path
+      modulePath: "$(GOPATH)/src/github.com/Azure/azure-container-networking"
+      dropgzVersion: $[ stagedependencies.setup.env.outputs['EnvironmentalVariables.dropgzVersion'] ]
+      cniVersion: $[ stagedependencies.setup.env.outputs['EnvironmentalVariables.cniVersion'] ]
+      npmVersion: $[ stagedependencies.setup.env.outputs['EnvironmentalVariables.npmVersion'] ]
+    dependsOn:
     - setup
     - publish
     jobs:
       - job: ${{ parameters.name }}
         displayName: Singletenancy AKS - (${{ parameters.name }})
         pool:
           name: $(BUILD_POOL_NAME_DEFAULT)
-          demands: 
+          demands:
           - agent.os -equals Linux
           - Role -equals $(CUSTOM_E2E_ROLE)
-        variables:
-          GOPATH: "$(Agent.TempDirectory)/go" # Go workspace path
-          GOBIN: "$(GOPATH)/bin" # Go binaries path
-          modulePath: "$(GOPATH)/src/github.com/Azure/azure-container-networking"
-          dropgzVersion: $[ stagedependencies.setup.env.outputs['EnvironmentalVariables.dropgzVersion'] ]
-          cniVersion: $[ stagedependencies.setup.env.outputs['EnvironmentalVariables.cniVersion'] ]
+        condition: always()
         steps:
           - template: e2e-step-template.yaml
             parameters:
               name: ${{ parameters.name }}
               clusterType: ${{ parameters.clusterType }}
               clusterName: ${{ parameters.clusterName }}
+              vmSize: ${{ parameters.vmSize }}
               arch: ${{ parameters.arch }}
               os: ${{ parameters.os }}
               k8sVersion: ${{ parameters.k8sVersion }}
               windowsOsSku: ${{ parameters.windowsOsSku }}
               os_version: ${{ parameters.os_version }}
               version: $(dropgzVersion)
-              cniVersion: $(cniVersion)
\ No newline at end of file
+              cniVersion: $(cniVersion)
+              scaleup: ${{ parameters.scaleup }}
+
+      - template: ../../cni/k8s-e2e/k8s-e2e-job-template.yaml
+        parameters:
+          sub: $(AZURE_TEST_AGENT_SERVICE_CONNECTION)
+          clusterName: ${{ parameters.clusterName }}
+          os: ${{ parameters.os }}
+          datapath: true
+          dns: true
+          portforward: true
+          loadBalancer: true
```

Comment:
Is there a case where we are don't want the data path or dns or portforward etc to be `false` ?

Reply:
The purpose of the template is to give us control and freedom of what tests we want to run. Generally, almost all future tests will require the cluster to have full connectivity, but having to run these tests every time the job template is called could be redundant.